tests/components/wallbox/test_lock.py

```diff
@@ -0,0 +1,127 @@
+"""Test Wallbox Lock component."""
+import json
+
+import pytest
+import requests_mock
+
+from homeassistant.components.lock import SERVICE_LOCK, SERVICE_UNLOCK
+from homeassistant.components.wallbox import CONF_LOCKED_UNLOCKED_KEY
+from homeassistant.const import ATTR_ENTITY_ID
+
+from tests.components.wallbox import (
+    entry,
+    setup_integration,
+    setup_integration_read_only,
+)
+from tests.components.wallbox.const import (
+    CONF_ERROR,
+    CONF_JWT,
+    CONF_MOCK_LOCK_ENTITY_ID,
+    CONF_STATUS,
+    CONF_TTL,
+    CONF_USER_ID,
+)
+
+authorisation_response = json.loads(
+    json.dumps(
+        {
+            CONF_JWT: "fakekeyhere",
+            CONF_USER_ID: 12345,
+            CONF_TTL: 145656758,
+            CONF_ERROR: "false",
+            CONF_STATUS: 200,
+        }
+    )
+)
+
+
+async def test_wallbox_lock_class(hass):
+    """Test wallbox lock class."""
+
+    await setup_integration(hass)
+
+    with requests_mock.Mocker() as mock_request:
+        mock_request.get(
+            "https://api.wall-box.com/auth/token/user",
+            json=authorisation_response,
+            status_code=200,
+        )
+        mock_request.put(
+            "https://api.wall-box.com/v2/charger/12345",
+            json=json.loads(json.dumps({CONF_LOCKED_UNLOCKED_KEY: False})),
+            status_code=200,
+        )
+
+        state = hass.states.get(CONF_MOCK_LOCK_ENTITY_ID)
+        assert state.state == "unlocked"
```

Comment:
This is good but it only tests the state after `setup_integration`. It would be interesting to test it after each service call, too.

Furthermore, `state` can be `None`. Just to be safe, please add an additional `assert state`.

Reply:
I don't see how that would work, in real life I send an unlock command to the api and afterwards refresh the status of the device. However, as I'm now mocking the requests, the status won't update (or I have to change the mock manually). So the test logic would be -> Mock (status Unlocked) -> Assert status -> Lock command (with Mock Request) -> Mock (status Locked) -> Assert status. I don't think I would really be testing something useful

Reply:
I don't think that would test anything, the status is retrieved from the API (which I'm mocking, so I'm mocking locked/unlocked state).

Reply:
Regarding state = None, I want to test that the lock platform doesn't get loaded, I find that it has a none state then. Is there a better way to check that the lock platform isn't loaded.

Reply:
I see, good point! In that case I would only recommend moving it up a little
```py
async def test_wallbox_lock_class(hass: HomeAssistant):
    """Test wallbox lock class."""

    await setup_integration(hass)

    state = hass.states.get(CONF_MOCK_LOCK_ENTITY_ID)
    assert state  # This is also new
    assert state.state == "unlocked"

    ...
```

> Regarding state = None, I want to test that the lock platform doesn't get loaded, I find that it has a none state then. Is there a better way to check that the lock platform isn't loaded?

I was only referring to the lines here. The `authentication_error` test is fine, I think. See `assert state` in the example above.

Reply:
OK, I changed this to the above example.